target/linux/ramips/patches-6.6/320-MIPS-add-support-for-buggy-MT7621S-core-detection.patch

```diff
@@ -56,16 +58,20 @@ Signed-off-by: Thomas Bogendoerfer <tsbogend@alpha.franken.de>
 +
 +	if (IS_ENABLED(CONFIG_SOC_MT7621)) {
 +		struct cpulaunch *launch;
++		unsigned long flags01;
 +
 +		/*
 +		 * Ralink MT7621S SoC is single core, but the GCR_CONFIG method
 +		 * always reports 2 cores. Check the second core's LAUNCH_FREADY
 +		 * flag to detect if the second core is missing. This method
-+		 * only works before the core has been started.
++		 * only works before starting the core. It also checks core0:1
++		 * to make sure the cores are not launched.
 +		 */
 +		launch = (struct cpulaunch *)CKSEG0ADDR(CPULAUNCH);
-+		launch += 2; /* MT7621 has 2 VPEs per core */
-+		if (!(launch->flags & LAUNCH_FREADY))
++		launch++; /* core0:1 */
++		flags01 = launch->flags;
++		launch++; /* core1:0 */
++		if ((flags01 & LAUNCH_FREADY) && !(launch->flags & LAUNCH_FREADY))
```

Comment:
It's not hard to detect the MT7621s variant by reading the syscon register.
```
	#include <asm/mach-ralink/mt7621.h>
	#define CHIP_REV_CORE_MASK		0x1
	#define CHIP_REV_CORE_SHIFT		17
	if (IS_ENABLED(CONFIG_SOC_MT7621)) {
		u32 sysc = __raw_readl(MT7621_SYSC_BASE + SYSC_REG_CHIP_REV);
		if (!((sysc >> CHIP_REV_CORE_SHIFT) & CHIP_REV_CORE_MASK))
			ncores = 1;
	}
```

Reply:
CPU0 revision is: 0001992f for single and dual core version.

Reply:
> CPU0 revision is: 0001992f for single and dual core version.

It's mips cpu revision, not soc revision. MTK SDK uses this way to start different mt7621 variants.
https://github.com/search?q=repo%3Agl-inet%2Fuboot-source-for_mtk%20RALINK_DUAL_CORE_FUN%20&type=code

Reply:
https://source.denx.de/u-boot/u-boot/-/blob/2b1c8d3b2da46ce0f7108f279f04bc66f1d8d09a/arch/mips/mach-mtmips/mt7621/init.c#L28-53

Reply:
@mrkiko Why you check `SYSC_REG_CHIP_NAME0` register instead of `SYSC_REG_CHIP_REV`?